src/components/NonTiledLayer.tsx

```diff
@@ -8,16 +8,26 @@ const NonTiledLayer: React.FC<{
   options: L.WMSOptions
   params?: { [key: string]: string }
 }> = ({ url, options, params }) => {
+  const [layer, setLayer] = useState<Layer>()
   const mapInstance = useMapInstance()
   const query = new URLSearchParams(params)
   const layerUrl = `${url}?${query}`
 
   useEffect(() => {
-    if (mapInstance !== null) {
-      L.nonTiledLayer.wms(layerUrl, options).addTo(mapInstance)
+    if (mapInstance !== null && L.nonTiledLayer) {
+      setLayer(L.nonTiledLayer.wms(layerUrl, options).addTo(mapInstance))
     }
   }, [mapInstance])
 
+  useEffect(
```

Comment:
Instead of creating a seperate `useEffect` for this is it not possible to return this de-registration from the pervious `useEffect`?